### ISSUE TITLE
Caching support

### ISSUE THREAD

**Issue by dclong** [pandas-profiling/pandas-profiling#561]
**Missing functionality**
Caching support seems to be missing. 

**Proposed feature**
Is it possible to enable caching when profiling a pandas DataFrame? 
Since pandas-profiling is for exploration,
users might repeatedly run profiling on datasets containing same subsets. 
Caching might help avoid unnecessary computation again.

**Comment by sbrugman** [comment]
@dclong Yes, pandas-profiling v3 will include a completely new execution system, that caches results.
